samples: nrf9160: Added Cloud Client sample (#1684)
NordicPlayground/fw-nrfconnect-nrf: samples/nrf9160/cloud_simple/README.rst

> .. _cloud_simple:

Reviewer: Suggestion: With reference to this suggestion: https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1684#pullrequestreview-339859690 , I also suggest cloud_client as the sample name. 
```suggestion
.. _cloud_client:
```

> nRF9160: Cloud Simple

Reviewer: ````Suggestion
nRF9160: Cloud Client
````

#####################

> Sample that connects to and communicates with a compatible cloud service using the respective firmware cloud backend. The sample connects via cellular network (LTE) and publishes a custom string in intervals or upon a button trigger, to the cloud service.

Reviewer: Please start every sentence on a new line. 
```Suggestion
This sample connects to, and communicates with a compatible cloud service using the respective cloud backend firmware. 
The sample connects via cellular network (LTE ) and publishes a custom string in intervals or upon a button trigger, to the cloud service.
````

Overview
********

> The ``cloud_simple`` sample demonstrates how to use the generic cloud API ``subsys/net/lib/cloud`` to interface with multiple cloud backends. The current version of the ``cloud_simple`` sample supports the following libraries as cloud backends:

Reviewer: ```Suggestion
The Cloud Client sample demonstrates how the generic cloud API ``subsys/net/lib/cloud`` can be used to interface with multiple cloud backends.
The current version of the sample supports the following libraries as cloud backends:
```

 -  :ref:`lib_nrf_cloud`
 -  :ref:`lib_aws_iot`

> To swap between cloud backends simply change the option ``CONFIG_CLOUD_BACKEND`` to match the configuration string of a compatible cloud backend. The identifying string for the different cloud backends are listed below.

Reviewer: ```Suggestion
To swap between cloud backends, change the option :option:`CONFIG_CLOUD_BACKEND` to match the configuration string of a compatible cloud backend.
The identifying string for the different cloud backends are listed in the following table.
```

 - **NRF Cloud**, configuration string ->"NRF_CLOUD"

>  - **AWS IoT**, configuration string -> "AWS_IOT"

Reviewer: Suggesting a table here:
````Suggestion
.. list-table::
   :header-rows: 1
   :align: center

   * - Cloud Backend
     - Configuration String
   * - NRF Cloud
     - NRF_CLOUD
   * - AWS IoT
     - AWS_IOT
````

Setup
*****

> For setting up the different cloud backends please see the corresponding documentations. Each cloud backend has spesific setup steps that must be executed before it can be used. These setup steps cover generating and flashing *certificates* used for the cloud connection, setting up the correct connection *endpoint* etc..

Reviewer: ````Suggestion
For setting up the different cloud backends, refer the documentation on :ref:`lib_nrf_cloud` and :ref:`lib_aws_iot` . 
Each cloud backend has specific setup steps that must be executed before it can be used.
These setup steps describes the following:
#. generating and flashing *certificates* used for the cloud connection
#. setting up the correct connection *endpoint* 
````

.. note::

>    The nRF9160 DK and Thingy:91 comes stock with certificates required for connection to NRF Cloud. For NRF Cloud no extra setup steps are required to use the ``cloud_simple`` sample. See `NRF Cloud web interface <https://nrfcloud.com/#/>`_ for more information.

Reviewer: ````Suggestion
   The nRF9160 DK and Thingy:91 comes with certificates required for connection to `nRF Cloud`_. 
   No extra steps are required to use the Cloud Client sample with nRF Cloud.
````

Configurations
**************

The configurations for the sample are located in ``cloud_simple/src/prj.conf``

 - ``CONFIG_CLOUD_BACKEND`` <- Determines the cloud backend to be used.
 - ``CONFIG_CLOUD_PUBLICATION_SEQUENTIAL`` <- Publish message to cloud sequentially.
 - ``CONFIG_CLOUD_PUBLICATION_BUTTON`` <- Publish message to cloud upon a button press.
 - ``CONFIG_CLOUD_MESSAGE`` <- Modifies the message published to the cloud service.
 - ``CONFIG_CLOUD_MESSAGE_PUBLICATION_INTERVAL`` <- Modifies how often the message should be published to the cloud service.

.. note::

>    To output data in the terminal window located in the `NRF Cloud web interface <https://nrfcloud.com/#/>`_ the data format must be in JSON format.

Reviewer: ````Suggestion
   To output data in the terminal window located in the `nRF Cloud`_ web interface, the data format must be in JSON format.
````

Functionality and Supported Technologies
****************************************

> The communicaton protocol supported by the sample is dependent on the cloud backend that is used. Please see the documentation for the individual cloud backend libraries.

Reviewer: ```suggestion
The communication protocol supported by the sample is dependent on the cloud backend that is used.
```
We already have links to the cloud backend documentation earlier in the documentation.

Functions
=========

> :cpp:func:`cloud_get_binding()`     : Binds to a desired cloud backend using a identification string.

Reviewer: ```suggestion
:cpp:func:`cloud_get_binding()`     : Binds to a desired cloud backend using an identification string.
```

> :cpp:func:`cloud_init()`            : Initializes the the cloud connection.

Reviewer: ```suggestion
:cpp:func:`cloud_init()`            : Initializes the cloud connection.
```

:cpp:func:`cloud_connect()`         : Connects to the cloud service.

> :cpp:func:`cloud_ping()`            : Pings the cloud service, should be called periodically.

Reviewer: ```suggestion
:cpp:func:`cloud_ping()`            : Pings the cloud service. This function should be called periodically.
```

> :cpp:func:`cloud_input()`           : Retrieve data from the cloud service.

Reviewer: ````Suggestion
:cpp:func:`cloud_input()`           : Retrieves data from the cloud service.
````

> :cpp:func:`cloud_send()`            : Send data to the cloud service.

Reviewer: ````Suggestion
:cpp:func:`cloud_send()`            : Sends data to the cloud service.
````

Cloud events used in the sample
===============================

> 

Reviewer: ```suggestion
Below are the major cloud events used in this sample and their short descriptions:
```

:cpp:enum:`CLOUD_EVT_CONNECTED`               : Connected to the cloud service.

:cpp:enum:`CLOUD_EVT_READY`                   : Cloud connection established and ready to be communicated with.

:cpp:enum:`CLOUD_EVT_DISCONNECTED`            : Disconnected from the cloud service.

> :cpp:enum:`CLOUD_EVT_DATA_RECEIVED`           : Receive data from the cloud service.

Reviewer: ````Suggestion
:cpp:enum:`CLOUD_EVT_DATA_RECEIVED`           : Received data from the cloud service.
````

.. note::

>    Not all functionality present in the generic cloud API are used by the different cloud backends.

Reviewer: ```Suggestion
   Not all functionalities present in the generic cloud API are used by the different cloud backends.
```

Requirements
************

* One of the following development boards:

>  * nRF9160 Development Kit board (PCA_20035)(Thingy:91)

Reviewer: ```suggestion
 * |Thingy91|
```

>  * nRF9160 Development Kit board (PCA_10090)(nRF9160 DK)

Reviewer: ```suggestion
 * |nRF9160DK|
```

> 

Reviewer: ```suggestion
* .. include:: /includes/spm.txt
```

Building and running
********************

> The sample can be found under ``samples/cloud_simple`` in the nRF Connect SDK folder structure.

Reviewer: ```suggestion
.. |sample path| replace:: :file:`samples/nrf9160/cloud_simple`
```
if we are changing the name of the sample to cloud_client, please remember to update here as well.

See :ref:`gs_programming` for information about how to build and program the application.

> The sample is configured to compile and run as a non-secure application on nRF91’s Cortex-M33. Therefore, it automatically includes the :ref:`secure_partition_manager` that prepares the required peripherals to be available for the application.

Reviewer: ```suggestion
.. include:: /includes/build_and_run_nrf9160.txt
```

> Current dependencies

Reviewer: ```Suggestion
Dependencies
````
